src/services/completions.ts

```diff
@@ -4299,6 +4298,34 @@ namespace ts.Completions {
             }
             return undefined;
         }
+
+    }
+
+    function getSingleLineReplacementSpanForImportCompletionNode(node: ImportDeclaration | ImportEqualsDeclaration | ImportSpecifier | Token<SyntaxKind.ImportKeyword> | undefined) {
+        if (!node) return undefined;
+        const top = (findAncestor(node, or(isImportDeclaration, isImportEqualsDeclaration)) || node) as ImportDeclaration | ImportEqualsDeclaration | Token<SyntaxKind.ImportKeyword>;

```

Comment:
Would be cool if `or` combined type predicates

also

```suggestion
        const top = (findAncestor(node, or(isImportDeclaration, isImportEqualsDeclaration)) ?? node) as ImportDeclaration | ImportEqualsDeclaration | Token<SyntaxKind.ImportKeyword>;
```

Reply:
I thought I had looked at that before and found it was impossible for some reason (???) but I guess it was only impossible to do it variadically. I added overloads for 2 and 3 unary type predicate functions. It eliminated one more type assertion on another call just like this, also written by me 😅